src/robusta/core/reporting/base.py

```diff
@@ -106,18 +106,28 @@ def __str__(self):
 
 class Filterable:
     @property
-    def attribute_map(self) -> Dict[str, str]:
+    def attribute_map(self) -> Dict[str, Union[str, Dict[str, str]]]:
         raise NotImplementedError
 
     def get_invalid_attributes(self, attributes: List[str]) -> List:
         return list(set(attributes) - set(self.attribute_map))
 
-    def attribute_matches(self, attribute: str, expression: Union[str, List[str]]) -> bool:
-        value = self.attribute_map[attribute]
-        if isinstance(expression, str):
-            return bool(re.match(expression, value))
+    def attribute_matches(self, attribute: str, expression: Union[str, List[str], Dict, List[Dict]]) -> bool:
+        value: Union[str, Dict[str, str]] = self.attribute_map[attribute]
+        if isinstance(expression, str) or isinstance(expression, Dict):
+            return Filterable.__value_match(value, expression)
         else:  # expression is list of values
-            return value in expression
+            return any([Filterable.__value_match(value, single_exp) for single_exp in expression])
+
+    @staticmethod
+    def __value_match(value: Union[str, Dict[str, str]], expression: Union[str, Dict]) -> bool:
+        if isinstance(value, str):
+            return bool(re.match(expression, value))
+        elif isinstance(value, Dict) and isinstance(expression, Dict):  # value is Dict[str, str], expression is a Dict
+            return expression.items() <= value.items()
```

Comment:
Why when both are dict this matches  expression.items() <= value.items() ?

Reply:
it checks that one dict is included in the other
We parse the string requirement into a dict, and then check for inclusion